File: reverse_team.py
import os
import glob

def reverse_in_file(file_path):
    with open(file_path, 'r', encoding='utf-8') as f:
        code = f.read()
    
    original = code
    code = code.replace(r'vady', 'voha')
    code = code.replace(r'vadyTeam', 'vohaTeam')
    code = code.replace(r'vadyStats', 'vohaStats')
    code = code.replace(r'Vady', 'Voha')
    code = code.replace('Вади', 'Вохи')
    code = code.replace('Вадя', 'Воха')

    if code != original:
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(code)
        print(f"Reversed: {file_path}")

for file_path in glob.glob('./src/**/*.ts', recursive=True) + glob.glob('./src/**/*.tsx', recursive=True):
    reverse_in_file(file_path)

if os.path.exists('./supabase-init.sql'):
    reverse_in_file('./supabase-init.sql')

print("Reverse replacement complete.")

File: reverse_team.js
const fs = require('fs');
const path = require('path');

function walkDir(dir, callback) {
  fs.readdirSync(dir).forEach(f => {
    let dirPath = path.join(dir, f);
    let isDirectory = fs.statSync(dirPath).isDirectory();
    isDirectory ? walkDir(dirPath, callback) : callback(dirPath);
  });
}

function reverseInFile(filePath) {
  if (!filePath.endsWith('.ts') && !filePath.endsWith('.tsx') && !filePath.endsWith('.sql')) return;
  
  let code = fs.readFileSync(filePath, 'utf8');
  let original = code;
  
  // Reversing Replacements
  code = code.replace(/\bvady\b/g, 'voha');
  code = code.replace(/\bvadyTeam\b/g, 'vohaTeam');
  code = code.replace(/\bvadyStats\b/g, 'vohaStats');
  code = code.replace(/\bVady\b/g, 'Voha');
  code = code.replace(/Вади/g, 'Вохи');
  code = code.replace(/Вадя/g, 'Воха');

  if (code !== original) {
    fs.writeFileSync(filePath, code, 'utf8');
    console.log(`Reversed: ${filePath}`);
  }
}

walkDir('./src', reverseInFile);
reverseInFile('./supabase-init.sql');

console.log('Reverse replacement complete.');
